psse/psse-model/src/main/java/com/powsybl/psse/model/PsseConstants.java

```diff
@@ -6,14 +6,45 @@
  */
 package com.powsybl.psse.model;
 
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
 /**
- *
  * @author Geoffroy Jamgotchian <geoffroy.jamgotchian at rte-france.com>
  */
 public final class PsseConstants {
 
     private PsseConstants() {
     }
 
-    public static final int SUPPORTED_VERSION = 33;
+    public enum PsseVersion {
+        VERSION_33(33),
+        VERSION_35(35);
+
+        private final int number;
+        private static final Map<Integer, PsseVersion> BY_NUMBER = new HashMap<>();
+
+        static {
+            for (PsseVersion v : PsseVersion.values()) {
+                BY_NUMBER.put(v.getNumber(), v);
+            }
+        }
+
+        private PsseVersion(int number) {
+            this.number = number;
+        }
+
+        public static PsseVersion fromNumber(int number) {
+            return BY_NUMBER.get(number);
+        }
+
+        public static Set<Integer> supportedVersions() {
+            return BY_NUMBER.keySet();
+        }
+
+        public int getNumber() {
+            return number;
+        }
+    }
```

Comment:
It is required to obtain the proper `Enum` value given an integer value read from the case. We want the _version_ to be passed along the different methods that will manage the conversion using a specific type instead of a generic `int`.